samples/WorkerBindingSamples/Cosmos/CosmosTriggerFunction.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Logging;
+
+namespace SampleApp
+{
+    // We cannot use trigger bindings with reference types because there is no way for the CosmosDB
+    // SDK to let us know the ID of the document that triggered the function; therefore we cannot create
+    // a client that is able to pull the triggering document.
```

Comment:
That's a fair point, we cannot support individual documents because we need the ID and there is no way to get the correct document that triggered the function. As for SDK types like CosmosClient, Database and Container - we don't need the ID or any document identifier so technically should be possible. However, I'm not sure this is a valid scenario since a) we're just aiming for parity right now https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-cosmosdb-v2-trigger?tabs=in-process%2Cfunctionsv2&pivots=programming-language-csharp#example and b) I'm not aware of scenarios where you would want to use a cosmos trigger and not use the triggering document in any way? 🤔

@mattchenderson - any thoughts on this? Should we be supporting CosmosClient for triggers?

Reply:
Had a chat offline with @mattchenderson about this and we agree that we do not need to support other types for a trigger. For now we are just aiming for in-proc parity and don't see a use case for using a cosmos client binding in a trigger. We can discuss further down the road if this is something we want to pursue